code/lib/builder-vite/src/vite-config.ts

```diff
@@ -52,7 +52,7 @@ export async function commonConfig(
   const sbConfig: InlineConfig = {
     configFile: false,
     cacheDir: 'node_modules/.cache/.vite-storybook',
-    root: path.resolve(options.configDir, '..'),
+    root: path.resolve('.storybook/', '..'),
```

Comment:
Don't think this works for all cases. Someone can set the config directory to a nested directory, but they can also set another directory entirely.

eg. `storybook dev --config-dir .storybook/nested`, `storybook dev --config-dir .storybook-config`, `storybook dev --config-dir config-storybook`, etc